cpp/src/arrow/filesystem/filesystem_test.cc

```diff
@@ -54,6 +54,17 @@ TEST(FileInfo, BaseName) {
   ASSERT_EQ(info.base_name(), "baz.qux");
 }
 
+TEST(CalculateSSECKeyMD5, Sanity) { 
+  ASSERT_FALSE(CalculateSSECKeyMD5("").ok());  // invalid base64
+  ASSERT_FALSE(CalculateSSECKeyMD5("%^H").ok());  // invalid base64
+  ASSERT_FALSE(CalculateSSECKeyMD5("INVALID").ok());  // invalid base64
+  ASSERT_FALSE(CalculateSSECKeyMD5("MTIzNDU2Nzg5").ok());  // invalid, the input key size not match
+  // valid case
+  auto result = CalculateSSECKeyMD5("1WH9aTJ0+Tn0NLbTMHZn9aCW3Li3ViAdBsoIldPCREw=");
+  ASSERT_TRUE(result.ok());  // valid case
+  ASSERT_STREQ(result->c_str(), "3HYIM58NCLwrIOdPpWnYwQ==");  // valid case
```

Comment:
Can we use `ASSERT_EQ()` here?

Reply:
Is there any other benefit to use the ASSERT_EQ()? for the EQ, if the result and baseline are not match, it would print the result which may be more friendly for developer to find the cause of the failure.

Reply:
I think that `ASSERT_STREQ()` is for C (string) not C++.
If we can use `std::string`, we should use `ASSERT_EQ()` instead of `ASSERT_STREQ()`.

I think that `ASSERT_EQ()` shows a developer friendly failure message too with `std::string`.